### PR TITLE
Make metav1.(Micro)?Time functions take pointers

### PR DESCRIPTION
**Pull request by gmarek**
Is there any reason for those functions not to be on pointers?

### CODE REVIEW

File: staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/micro_time.go [hunk 1]
```diff
@@ -74,22 +74,22 @@ func (t *MicroTime) IsZero() bool {
 }
 
 // Before reports whether the time instant t is before u.
-func (t MicroTime) Before(u MicroTime) bool {
+func (t *MicroTime) Before(u MicroTime) bool {
```
Comment (sttts) [inline staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/micro_time.go]: non-pointer arguments are odd now, looks asymmetric with this change.

Comment (gmarek) [inline staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/micro_time.go]: It should have been an interface in first place...

Comment (gmarek) [inline staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/micro_time.go]: Changed it to take ptr argument and updated the code everywhere in main repo.

**Comment by sttts** [comment]
Looks good to me. @deads2k which group do you belong to? Do non-pointer receivers hurt your eyes?

**Comment by deads2k** [comment]
> Looks good to me. @deads2k which group do you belong to? Do non-pointer receivers hurt your eyes?

No opinion either way.

/approve

**Comment by gmarek** [comment]
@sttts - PTAL, all tests passed.

**Comment by sttts** [comment]
@smarterclayton approved?

/assign @smarterclayton

**Comment by smarterclayton** [comment]
I think these were non-pointers because this is a struct wrapper and we were being consistent with Go time.  I don't care one way or another, but Go times were treated as value types at one point.

**Comment by gmarek** [comment]
Ping @smarterclayton

**Pull request merged**